packages/astro/src/core/create-vite.ts

```diff
@@ -102,25 +102,13 @@ export async function createVite(
 		},
 	});
 
-	const viteCustomLogger: ViteLogger = {
-		...vite.createLogger('warn'),
-		// All error log messages are also thrown as real errors,
-		// so we can safely ignore them here and let the error handler
-		// log them for the user instead.
-		error: (msg) => logger.debug('vite', 'ERROR ' + msg),
-		// Warnings are usually otherwise ignored by Vite, so it's
-		// important that we catch and log them here.
-		warn: (msg) => logger.warn('vite', msg),
-	};
```

Comment:
I overlooked this part while reviewing https://github.com/withastro/astro/pull/9105.

Passing `vite.createLogger('warn')` directly has the issue where users are unable to set a different `logLevel` (even for us), which caused the client build to not log info messages anymore.

The `error` function here could also suppress errors that we don't want to suppress. Vite may sometime call `logger.error()` but doesn't throw any errors.

Reply:
This was intentional! Every error that I encountered with Vite was also a thrown error, which caused double-logging on almost every error we encounter since Vite is the first one to hit it. 

For example, trying to import a file that doesn't exist results in different 3 error messages now instead of 1, because Vite logs two redundant error messages:

<img width="1071" alt="Screen Shot 2023-11-29 at 4 03 57 PM" src="https://github.com/withastro/astro/assets/622227/0184b56f-c2fc-42ed-b074-fe00b571b0a5">

To summarize the approach I went with:
- Our AstroError information is higher fidelity than Vite's logs, more helpful
- Our handling of Vite errors is good enough at turning them into AstroError
- If all relevant Vite error logs are also reported as AstroErrors, then Vite errors actually are just extra redundent info, useful for debugging but not needed in the console.
- Any Vite errors that aren't thrown / can't be handled by Astro can be handled individually in this logger and turned into error logs. But the model would be an  allowlist / opt-in instead of default logging all Vite errors to the console.

Reply:
Also, you should rebase/erge onto `next` which now includes https://github.com/withastro/astro/pull/9129 changes

Reply:
Thanks! Yeah I made sure that for Vite-turned-Astro errors, we also skip them at https://github.com/withastro/astro/blob/6e46fc0194601008f73614ae3248195b9d265444/packages/astro/src/core/logger/vite.ts#L75-L76

Last I tested it works locally so that it only logged once, so I hope it addresses the concern here.

Reply:
After rebasing, it seems likeit's showing what you have in your screenshot, gonna check this again

Reply:
Just fixed them in the latest commit, it looks like this now:
<img width="634" alt="image" src="https://github.com/withastro/astro/assets/34116392/72195d5b-560c-4fe4-82ee-f7a6f6bbaa82">

Reply:
Nice! I can reproduce the same, feel much better about that then.